trie/typing.py

```diff
@@ -0,0 +1,35 @@
+import enum
+
+from eth_utils import (
+    is_list_like,
+)
+
+
+class Nibble(enum.IntEnum):
+    Hex0 = 0
+    Hex1 = 1
+    Hex2 = 2
+    Hex3 = 3
+    Hex4 = 4
+    Hex5 = 5
+    Hex6 = 6
+    Hex7 = 7
+    Hex8 = 8
+    Hex9 = 9
+    HexA = 0xA
+    HexB = 0xB
+    HexC = 0xC
+    HexD = 0xD
+    HexE = 0xE
+    HexF = 0xF
+
+    def __repr__(self):
+        return hex(self.value)
+
+
+class Nibbles(tuple):
+    def __new__(cls, nibbles):
+        if not is_list_like(nibbles):
+            raise TypeError(f"Must pass in a tuple of nibbles, but got {nibbles!r}")
+        else:
+            return tuple.__new__(cls, (Nibble(maybe_nibble) for maybe_nibble in nibbles))
```

Comment:
Hm, I don't think there's a good way to catch an exception from `Nibble` and clarify it, because the same exception would be raised by the following snippets:

```py
for n in b'A':
  Nibble(n)

for n in (0x41, ):
  Nibble(n)
```

So I just added a tiny wrapper for a series of nibbles, plus a test.

Reply:
I was just thinking:

```python
try:
    ...  # code that does `Nibble(value)`
except TypeError as err:
    raise TypeError("Some message that is more informative") from err
```